### DISCUSSION TITLE
Use an old runner image version

### DISCUSSION THREAD

**Discussion by dantebarba** [community/community#160655] [Actions] [answered]
Today our team was deploying a new version of our testing app when to our surprise an unexpected error came up. We are currently unable to determine what is the cause of the error since building the project from our local environments and then deploying it to our GCP node works ok, but not from Github actions.
After digging a little bit we've noticed that the last successful deployment was on April 24th, and 2 days after the Ubuntu image was updated.

This was the old version that we were using: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20230417.1

This is the new one: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20230426.1

We suspect that the new docker version from which the image is being build it's behind this issue. Unfortunately we cannot test this because there is no way to rollback to a different runner version? Or we might be mistaken.

Is there a solution were we can copy the build environment as much as possible to isolate the issue? We'd like to have the possibility to specify the exact Runner Image version.

**Comment by artsiom-abakumov** [comment]
I faced similar issue with kind cluster started to fail without clear reason and dates are clsoe to recent image change...

**Comment by v-sergei-pyshnoi** [comment, marked as answer]
Hello @dantebarba . Unfortunately it`s impossible to specify older version of runner-images during  workflow. The only thing a can suggest its only investigate release notes or try to build image from previous release using packer.

**Reply by dantebarba** [reply]
Such a bummer. Fortunately we've tracked the issue to an incorrect dependency version, but in case that an issue with the runner image actually occurs it would be a challenge to solve.

**Reply by MostefaKamalLala** [reply]
I am facing a similar issue with the updated installed software that are not compatible with my workload. Another option would be to tu setup the dependency at the right version on the new runner until you fix your issue.

**Reply by fernandoaleman** [reply]
I am now facing a similar issu. Since the docker packages were updated this week, our chef cookbook tests now fail because systemd no longer seems to be enabled in docker with the newer verisons.

**Comment by Zibri** [comment]
that's too bad. I think there should be a mechanism to specify the version. It often happens that an update breaks compilation scripts or worse.
Alternatively you could create "immutable" image along with the updated one. In this way once scripts work with the immutable version they will always work.

**Comment by avoitenko-logitech** [comment]
Inability to use specific images make usage of Github Actions pretty useless for creation of production builds, because everything might break suddenly and usually in the most inconvenient moment.

**Comment by KillerCodeMonkey** [comment]
run in an issue with electron + electron-builder... macos signing is failing now out of sudden..
https://github.com/electron-userland/electron-builder/issues/8306

no glue how to solve it. I am not able to setup a large macos runner to check, if now there is limited  memory usage of a node process or what ever happens..

everything worked with https://github.com/actions/runner-images/releases/tag/macos-14-arm64%2F20240616.1

and stopped with all later images... even non arm or macos-12|13... and i have no chance to check or validate if it is still working on older images in github actions.

Currently we are not able to sign our apps with github actions...
